solidity/contracts/fully-backed/FullyBackedECDSAKeepFactory.sol

```diff
@@ -236,11 +255,32 @@ contract FullyBackedECDSAKeepFactory is
         return bonding.isAuthorizedForOperator(_operator, address(this));
     }
 
+    /// @notice Bans members of a calling keep in all associated sortition pools
+    /// for every registered application.
+    /// @dev The function can be called only by a keep created by this factory.
+    function banKeepMembers() public onlyKeep() {
+        FullyBackedECDSAKeep keep = FullyBackedECDSAKeep(msg.sender);
+
+        address[] memory members = keep.getMembers();
+
+        for (uint256 i = 0; i < members.length; i++) {
+            address operator = members[i];
+
+            for (uint256 j = 0; j < applications.length; j++) {
```

Comment:
We may get into problems here. There is a chance adversary may create a bunch of applications and this call will fail with out-of-gas so reporting frauds will not be possible. Instead of banning from all sortition pools, we should ban from the specific sortition pool.

I believe it should be farily easy as the owner of the keep is the application, right? So we can get the application address from the keep. Worth confirming...

Reply:
This was also my concern, I'm glad you brought it up. Unfortunatelly solving this is not that straightforward, owner of a keep is a Deposit not an Application. I'll try to look for a solution.

Reply:
Well, even if the owner was an application, assuming owner == application would suck for the API because owner is a parameter and the application may chose anyone as an owner which is actually happening here. I think the easiest option is to add a map `keep -> application` in `FullyBackedECDSAKeepFactory`.

Reply:
> I think the easiest option is to add a map keep -> application in FullyBackedECDSAKeepFactory.

Yep, I'm already working on this.